### PR TITLE
Registration boxplots

### PR DESCRIPTION
**Pull request by stephaniege**
Boxplots are mostly in place, minus the fake data being used.

### CODE REVIEW

**Comment by seshness** [comment]
this will have conflicts with #110

File: app/assets/javascripts/superuser.js [hunk 1]
```diff
@@ -1,2 +1,218 @@
...
+    .range([0,chart.width()]);
+
+  chart.xscale(xs);
+
+  var x_axis = d3.svg.axis()
+    .scale(xs)
+    .ticks(d3.time.days,1)
+    .orient('bottom');
+
+  var svg = d3.select("body").selectAll("svg")
```
Comment (seshness) [inline app/assets/javascripts/superuser.js]: This is problematic.  We will have multiple `svg` elements on the page. Create an `svg` element in `superuser.html.haml` with an `id` attribute, and select that `svg` element specifically here.

File: app/assets/javascripts/superuser.js [hunk 2]
```diff
@@ -1,2 +1,218 @@
...
+Superuser.prototype.init = function() {
+  Raxa.prototype.init.call(this);
+  this.initEncountersHistory();
+};
+
+Superuser.prototype.initEncountersHistory = function() {
+
+  var w = 520,
+      h = 300,
+      m = [10, 100, 20, 50];
```
Comment (danwang) [inline app/assets/javascripts/superuser.js]: Can we use names for this?

{top : X, right: Y, bottom: Z, ...}

File: app/assets/javascripts/superuser.js [hunk 3]
```diff
@@ -1,2 +1,218 @@
...
+
+  var chart = d3.box()
+              .width(w - m[1] - m[3])
+              .height(h - m[0] - m[2]);
+
+  var max = -Infinity,
+      min = Infinity;
+  $.each(data.map (function(x)  { console.log(x); return x.data;}), function(i, d) {
```
Comment (danwang) [inline app/assets/javascripts/superuser.js]: console.log

File: app/assets/javascripts/superuser.js [hunk 4]
```diff
@@ -1,2 +1,218 @@
...
+    max = d.max > max ? d.max : max;
+    min = d.min < min ? d.min : min;
```
Comment (danwang) [inline app/assets/javascripts/superuser.js]: Why not just have the unpacking inside the `.each`?

Comment (danwang) [inline app/assets/javascripts/superuser.js]: Also, check out `Math.max`

File: app/assets/javascripts/superuser.js [hunk 5]
```diff
@@ -1,2 +1,218 @@
...
+    .domain([min, max])
+    .range([chart.height(), 0]);
+
+  chart.yscale(ys);
+
+  var y_axis = d3.svg.axis()
+    .scale(ys)
+    .orient('left');
+
+  //var xs = d3.scale.ordinal()
```
Comment (danwang) [inline app/assets/javascripts/superuser.js]: ?

File: app/assets/javascripts/superuser.js [hunk 6]
```diff
@@ -1,2 +1,218 @@
...
+  var dates = data.map(function(x) { return x.date; });
+  var date_sort_asc = function (date1, date2) {
+    if (date1 > date2) return 1;
+    if (date1 < date2) return -1;
+    return 0;
+  };
+  dates.sort(date_sort_asc);
+  console.log(dates);
```
Comment (danwang) [inline app/assets/javascripts/superuser.js]: console.log

File: app/assets/javascripts/superuser.js [hunk 7]
```diff
@@ -1,2 +1,218 @@
...
+
+  var y_axis = d3.svg.axis()
+    .scale(ys)
+    .orient('left');
+
+  //var xs = d3.scale.ordinal()
+  var dates = data.map(function(x) { return x.date; });
+  var date_sort_asc = function (date1, date2) {
+    if (date1 > date2) return 1;
+    if (date1 < date2) return -1;
```
Comment (danwang) [inline app/assets/javascripts/superuser.js]: Is this function necessary? It seems to be translating a boolean into 1/-1.

File: app/assets/javascripts/superuser.js [hunk 8]
```diff
@@ -1,2 +1,218 @@
...
+
+  var svg = d3.select("body").selectAll("svg")
+    .data([data])
+    .enter().append("svg")
+      .attr("class", "box")
+      .attr("width", w + m[1] + m[3])
+      .attr("height", h + m[0] + m[2]);
+
+  svg.append("g")
+    .attr("transform", "translate(" + m[1] + "," + m[2] + ")")
```
Comment (danwang) [inline app/assets/javascripts/superuser.js]: m.up/right/left/down

File: app/assets/javascripts/superuser.js (hunk 7 above)
Comment (danwang) [inline app/assets/javascripts/superuser.js]: How about

`dates.sort(function(d1, d2) { return d2 - d1; })`

File: app/assets/javascripts/superuser.js [hunk 9]
```diff
@@ -1,2 +1,218 @@
...
+        .attr("class", "median")
+        .attr("x1", -boxwidth/2)
+        .attr("y1", yscale)
+        .attr("x2", boxwidth/2)
+        .attr("y2", yscale);
+
+      var whisker = g.selectAll("line.whisker")
+        .data([d.min, d.max]);
+
+      whisker.enter().append("line") //???
```
Comment (danwang) [inline app/assets/javascripts/superuser.js]: //???

**Comment by erictzeng** [comment]
I think this is outdated.

**Pull request closed without merging**